build/AppveyorBuildFunctions.ps1

```diff
@@ -0,0 +1,130 @@
+function Build-Solution
+{
+    Write-Host "Building Relay projects"
+
+    dotnet restore
+
+    # $? Returns True or False value indicating whether previous command ended with an error.
+    # This is used to throw an error that will cause the AppVeyor process to fail as expected.
+    if (-not $?)
+    {
+        throw "Package restore failed."
+    }
+
+    dotnet build src/Microsoft.Azure.Relay/project.json
+
+    if (-not $?)
+    {
+        throw "Build failed."
```

Comment:
Maybe change these error messages to say _which_ build failed?  (Since there are two places which 'throw "Build failed"').